admin/module/tests/README.md

```diff
@@ -1,62 +1,81 @@
 # Running Module Tests
 
-This is the quick-start to CodeIgniter testing. Its intent is to describe what 
-it takes to set up your module and get it ready to run unit tests. 
-It is not intended to be a full description of the test features that you can 
+This is the quick-start to CodeIgniter testing. Its intent is to describe what
+it takes to set up your module and get it ready to run unit tests.
+It is not intended to be a full description of the test features that you can
 use to test your module. Those details can be found in the documentation.
 
 ## Resources
+
 * [CodeIgniter 4 User Guide on Testing](https://codeigniter4.github.io/userguide/testing/index.html)
-* [PHPUnit docs](https://phpunit.readthedocs.io/en/8.5/index.html)
+* [PHPUnit docs](https://phpunit.de/documentation.html)
+* [Any tutorials on Unit testing in CI4?](https://forum.codeigniter.com/showthread.php?tid=81830)
 
 ## Requirements
 
-It is recommended to use the latest version of PHPUnit. At the time of this 
-writing we are running version 8.5.13. Support for this has been built into the 
-**composer.json** file that ships with CodeIgniter and can easily be installed 
+It is recommended to use the latest version of PHPUnit. At the time of this
+writing we are running version 9.x. Support for this has been built into the
+**composer.json** file that ships with CodeIgniter and can easily be installed
 via [Composer](https://getcomposer.org/) if you don't already have it installed globally.
 
-	> composer install
+```console
+> composer install
+```
 
-If running under OS X or Linux, you can create a symbolic link to make running tests a touch nicer.
+If running under macOS or Linux, you can create a symbolic link to make running tests a touch nicer.
 
-	> ln -s ./vendor/bin/phpunit ./phpunit
+```console
+> ln -s ./vendor/bin/phpunit ./phpunit
+```
 
-You also need to install [XDebug](https://xdebug.org/index.php) in order
-for code coverage to be calculated successfully.
+You also need to install [XDebug](https://xdebug.org/docs/install) in order
+for code coverage to be calculated successfully. After installing `XDebug`, you must add `xdebug.mode=coverage` in the **php.ini** file to enable code coverage.
```

Comment:
in phpunit.xml can be configured as well iirc:

```xml
<env name="XDEBUG_MODE" value="coverage"/>
```

Reply:
No, it does not work.
The code `<env name="XDEBUG_MODE" value="coverage"/>` is already described in CI4's `phpunit.xml.dist`.

Reply:
If I remove `xdebug.mode=coverage` in php.ini:
```console
$ vendor/bin/phpunit tests/system/CodeIgniterTest.php 
PHPUnit 9.5.21 #StandWithUkraine

Runtime:       PHP 8.0.21 with Xdebug 3.1.5
Configuration: /Users/kenji/work/codeigniter/CodeIgniter4/phpunit.xml

Code coverage needs to be enabled in php.ini by setting 'xdebug.mode' to 'coverage'
```

But the following works:
```console
$ XDEBUG_MODE=coverage vendor/bin/phpunit tests/system/CodeIgniterTest.php 
PHPUnit 9.5.21 #StandWithUkraine

Runtime:       PHP 8.0.21 with Xdebug 3.1.5
Configuration: /Users/kenji/work/codeigniter/CodeIgniter4/phpunit.xml

........................                                          24 / 24 (100%)

Time: 00:05.303, Memory: 46.00 MB

OK (24 tests, 26 assertions)

Generating code coverage report in Clover XML format ... done [00:10.618]
```

Reply:
Ok 👍